### PR TITLE
feat(core): Read-Only FiloDB capability for Querying Downsampled Data

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

This PR adds new Memstore implementation to support serving downsampled data from Cassandra without caching it in off-heap memory. This read-only store does not perform ingestion.

`Memstore` BTW, should be renamed to just `TimeSeriesStore`.

PR also includes unit test that writes raw data to cassandra, runs downsampler batch job and reads downsampled data using PagedReadablePartition

### CODE REVIEW

File: spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala [hunk 1]
```diff
@@ -14,10 +14,11 @@ import filodb.cassandra.FiloSessionProvider
 import filodb.cassandra.columnstore.CassandraColumnStore
 import filodb.core.{DatasetRef, ErrorResponse, Instance}
 import filodb.core.binaryrecord2.RecordSchema
-import filodb.core.downsample.{ChunkDownsampler, DoubleChunkDownsampler, HistChunkDownsampler, TimeChunkDownsampler}
+import filodb.core.downsample._
```
Comment (broneill) [inline spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala]: I generally prefer to see this change going the other way -- fewer wildcards. I don't care too much, however.

File: core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesStore.scala [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  override def recoverStream(dataset: DatasetRef, shard: Int,
+                             stream: Observable[SomeData],
+                             startOffset: Long, endOffset: Long, checkpoints: Map[Int, Long],
+                             reportingInterval: Long): Observable[Long] = throw new UnsupportedOperationException()
+
+  override def numPartitions(dataset: DatasetRef, shard: Int): Int = throw new UnsupportedOperationException()
+
+  override def numRowsIngested(dataset: DatasetRef, shard: Int): Long = throw new UnsupportedOperationException()
+
+  override def latestOffset(dataset: DatasetRef, shard: Int): Long = throw new UnsupportedOperationException()
```
Comment (broneill) [inline core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesStore.scala]: Isn't ??? the same as throwing this exception?

File: spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala (hunk 1 above)
Comment (vishramachandran) [reply spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala]: It was intellij auto imports at play. It introduces wildcard after a certain number of whitelisted imports.

File: core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesStore.scala (hunk 2 above)
Comment (vishramachandran) [reply core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesStore.scala]: Both throw exceptions, but different ones. `???` is technically not-implemented-error.

**Pull request merged**